integration-tests/helpers.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+const express = require('express')
+const bodyParser = require('body-parser')
+const msgpack = require('msgpack-lite')
+const codec = msgpack.createCodec({ int64: true })
+const EventEmitter = require('events')
+const { fork } = require('child_process')
+const http = require('http')
+
+class FakeAgent extends EventEmitter {
+  constructor (port = 0) {
+    super()
+    const agent = express()
```

Comment:
Most of the constructor could be moved to a method.

Reply:
It could, but why?

Reply:
Since there is a close method, I figured it should have a corresponding start or init or something like that so that it can be reused. Although we do it in many places, side-effects in constructors should also generally be avoided. Definitely not a blocker if it makes usage easier to keep it as-is.

Reply:
Switched to `start` and `stop` methods.